frontend/frontend.go

```diff
@@ -13,7 +13,7 @@ import (
 )
 
 type Frontend interface {
-	Solve(ctx context.Context, llb FrontendLLBBridge, opt map[string]string) (*Result, error)
+	Solve(ctx context.Context, llb FrontendLLBBridge, opt map[string]string, inputs map[string]llb.State) (*Result, error)
```

Comment:
I'd prefer `*pb.Definition` type more in here, as `llb` pkg is just a client helper package for generating proto and doesn't really belong to backend API definitions. But let me know if this complicates anything.